backend/src/components/edx/exchange.js

```diff
@@ -238,6 +238,29 @@ async function markAs(req, res) {
   }
 }
 
+async function markAsClosed(req, res) {
+  const token = utils.getBackendToken(req);
+  if (!token) {
+    return res.status(HttpStatus.UNAUTHORIZED).json({
+      message: 'No access token'
+    });
+  }
+  try {
+    const currentExchange = await getData(token, config.get('server:edx:exchangeURL') + `/${req.params.secureExchangeID}`);
+    currentExchange.secureExchangeStatusCode = 'CLOSED';
+    currentExchange.createDate = null;
+    currentExchange.updateDate = null;
+
```

Comment:
Please add an additional check here that the logged in user has the correct permission & school access. You can use the `activeInstituteIdentitfier` & `activeInstitutePermissions` to confirm. Just check it against the `currentExchange` you pulled.